src/abstractdatatable/reshape.jl

```diff
@@ -216,7 +214,8 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
             payload[j][i]  = valuecol[k]
         end
     end
-    insert!(payload, 1, NullableArray(levels(refkeycol)), _names(dt)[rowkey])
+    col = typeof(similar_nullable(dt[rowkey], 1))(levels(refkeycol))
```

Comment:
The principle is good, but I don't really like that way of using `typeof`, since in theory arrays do not necessarily have a constructor like this. Better do:
```julia
levs = levels(levels(refkeycol))
col = similar_nullable(dt[rowkey], length(levs))
copy!(col, levs)
```